packages/nodes-base/nodes/N8nApi/WorkflowDescription.ts

```diff
@@ -0,0 +1,393 @@
+ * data, if 'removeReadOnlyFields' is set to true.
+ *
+ * NOTE! This expects the requestOptions.body to already be set as an object,
+ * so take care to first call parseAndSetBodyJson().
+ */
+const maybeRemoveReadOnlyFields: PreSendAction = async function (
+	this: IExecuteSingleFunctions,
+	requestOptions: IHttpRequestOptions,
+): Promise<IHttpRequestOptions> {
+	const shouldRemove = this.getNodeParameter('removeReadOnlyFields') === true;
+	if (shouldRemove) {
+		requestOptions.body = omit(requestOptions.body as {}, READ_ONLY_FIELDS);
+	}
+	return requestOptions;
+};
+
+export const workflowOperations: INodeProperties[] = [
+	{
+		displayName: 'Operation',
+		name: 'operation',
+		type: 'options',
+		noDataExpression: true,
+		default: 'getAll',
+		displayOptions: {
+			show: {
+				resource: ['workflow'],
+			},
+		},
+		options: [
+			{
+				name: 'Activate',
+				value: 'activate',
+				action: 'Activate a workflow',
+				routing: {
+					request: {
+						method: 'POST',
+						url: '=/workflows/{{ $parameter.workflowId }}/activate',
+					},
+				},
+			},
+			{
+				name: 'Create',
+				value: 'create',
+				action: 'Create a workflow',
+				routing: {
+					request: {
+						method: 'POST',
+						url: '/workflows',
+					},
+				},
+			},
+			{
+				name: 'Deactivate',
+				value: 'deactivate',
+				action: 'Deactivate a workflow',
+				routing: {
+					request: {
+						method: 'POST',
+						url: '=/workflows/{{ $parameter.workflowId }}/deactivate',
+					},
+				},
+			},
+			{
+				name: 'Delete',
+				value: 'delete',
+				action: 'Delete a workflow',
+				routing: {
+					request: {
+						method: 'DELETE',
+						url: '=/workflows/{{ $parameter.workflowId }}',
+					},
+				},
+			},
+			{
+				name: 'Get',
+				value: 'get',
+				action: 'Get a workflow',
+				routing: {
+					request: {
+						method: 'GET',
+						url: '=/workflows/{{ $parameter.workflowId }}',
+					},
+				},
+			},
+			{
+				name: 'Get Many',
+				value: 'getAll',
+				action: 'Get many workflows',
+				routing: {
+					request: {
+						method: 'GET',
+						url: '/workflows',
+					},
+					output: {
+						postReceive: [
+							{
+								type: 'rootProperty',
+								properties: {
+									property: 'data',
+								},
+							},
+						],
+					},
+				},
+			},
+			{
+				name: 'Update',
+				value: 'update',
+				action: 'Update a workflow',
+				routing: {
+					request: {
+						method: 'PUT',
+						url: '=/workflows/{{ $parameter.workflowId }}',
+					},
+				},
+			},
+		],
+	},
+];
+
+const activateOperation: INodeProperties[] = [
+	{
+		displayName: 'Workflow ID',
+		name: 'workflowId',
+		type: 'string',
+		default: '',
+		required: true,
+		displayOptions: {
+			show: {
+				resource: ['workflow'],
+				operation: ['activate'],
+			},
+		},
+		description: 'ID of the workflow to activate',
+	},
+];
+
+const createOperation: INodeProperties[] = [
+	{
+		displayName: 'Workflow Object',
+		name: 'workflowObject',
+		type: 'string',
```

Comment:
Noticed I have differing types for create and update. Probably doesn't change the behavior, but I'll check...